pkg/entities/set.go

```diff
@@ -59,16 +58,7 @@ type set struct {
 	records []Record
 }
 
-func NewSet(buffer *bytes.Buffer) Set {
-	return &set{
-		buffer:  buffer,
-		currLen: 0,
-		setType: Undefined,
-		records: make([]Record, 0),
-	}
-}
-
-func (s *set) CreateNewSet(setType ContentType, templateID uint16) error {
+func NewSet(setType ContentType, templateID uint16) Set {
 	// Create the set header and append it
```

Comment:
Do we need set header for collector/decode messages too?

Reply:
we don't need set header or set buffer when decoding message. I will add a `isDecoding` field in set to handle that then. Thanks!